### PR TITLE
Generate the presigned url when the task launches instead of generati…

### PR DESCRIPTION
**Pull request by TheoPascoli**
…ng it in database when the file is created

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*

**What is the current behavior?** *(You can also link to an open issue here)*

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change?** *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: src/main/resources/application.yml [hunk 1]
```diff
@@ -49,3 +49,10 @@ scheduler:
   start-hour: 0
   end-hour: 20
   frequency-in-minutes: 1
+
+minio-adapter:
+  bucket: gridcapa
+  base-path: /SWE/D2CC
+  access-key: gridcapa
+  secret-key: gridcapa
+  url: http://localhost:9000
```
Comment (schwitzguebel) [inline src/main/resources/application.yml:58]: New line at end of file please :)

File: src/test/resources/application.yml [hunk 2]
```diff
@@ -15,3 +15,10 @@ scheduler:
   start-hour: 0
   end-hour: 20
   frequency-in-minutes: 1
+
+minio-adapter:
+  bucket: gridcapa
+  base-path: /SWE/D2CC
+  access-key: gridcapa
+  secret-key: gridcapa
+  url: http://localhost:9000
```
Comment (schwitzguebel) [inline src/test/resources/application.yml:24]: New line at end of file please :)

**Review by schwitzguebel — APPROVED** [review]

**Review by schwitzguebel — APPROVED** [review]

File: pom.xml [hunk 3]
```diff
@@ -14,8 +14,9 @@
 
 	<properties>
 		<awaitability.version>3.1.6</awaitability.version>
+		<gridcapa.starter.minio.adapter.version>1.0.0</gridcapa.starter.minio.adapter.version>
 		<logstash.version>4.9</logstash.version>
-		<task-manager.version>1.4.0</task-manager.version>
+		<task-manager.version>1.12.0</task-manager.version>
```
Comment (WalAmeni) [inline pom.xml:19]: I think we should increment the version of task manager with the new one after release!

File: src/main/java/com/farao_community/farao/gridcapa/job_launcher/JobLauncherService.java [hunk 4]
```diff
@@ -85,6 +93,25 @@ public boolean launchJob(String timestamp) {
...
     }
 
+    TaskDto replaceFilenameWithPresignedUrls(TaskDto taskDto) {
+        List<ProcessFileDto> newInput = new ArrayList<>();
+        for (ProcessFileDto input : taskDto.getInputs()) {
+            ProcessFileDto processFileDto = new ProcessFileDto(input.getFileType(),
+                    input.getProcessFileStatus(),
+                    input.getFilename(),
+                    input.getLastModificationDate(),
+                    minioAdapter.generatePreSignedUrlFromFullMinioPath(input.getFileUrl(), 1));
```
Comment (WalAmeni) [inline src/main/java/com/farao_community/farao/gridcapa/job_launcher/JobLauncherService.java:103]: That's working but I don't think it's clean to modify ProcessFileDto here. This is also linked with my comment in the task manager PR, it's not pretty to use the same variable for two different meaning. In the task manager the FileUrl correspond to the object key (Minio path) and here is replaced by an url. I think also it's better to use the minio.generatePresignedUrlFromFullMinioPath method in the process adapter when it creates the process request. We can discuss about it !

**Review by WalAmeni — CHANGES_REQUESTED** [review]

**Pull request closed without merging**